File: extensions/jashing-cvs/src/main/java/com/github/avarabyeu/jashing/integration/vcs/AbstractVCSClient.java
package com.github.avarabyeu.jashing.integration.vcs;

import javax.annotation.Nonnull;
import java.time.Instant;
import java.util.Map;

/**
 * Some basic methods for {@link com.github.avarabyeu.jashing.integration.vcs.VCSClient}
 * Just delegate calls to appropriate methods
 *
 * @author Andrey Vorobyov
 */
public abstract class AbstractVCSClient implements VCSClient {

    @Override
    public Map<String, Long> getCommitsPerUser(@Nonnull Instant from) {
        return getCommitsPerUser(from, null);
    }

    @Override
    public long getCommitsForPeriod(@Nonnull Instant from) {
        return getCommitsForPeriod(from, null);
    }
}

File: extensions/jashing-cvs/src/main/java/com/github/avarabyeu/jashing/integration/vcs/AbstractVCSClient.kt
package com.github.avarabyeu.jashing.integration.vcs

import java.time.Instant

/**
 * Some basic methods for [com.github.avarabyeu.jashing.integration.vcs.VCSClient]
 * Just delegate calls to appropriate methods

 * @author Andrey Vorobyov
 */
abstract class AbstractVCSClient : VCSClient {

    override fun getCommitsPerUser(from: Instant): Map<String, Long> {
        return getCommitsPerUser(from, null)
    }

    override fun getCommitsForPeriod(from: Instant): Long {
        return getCommitsForPeriod(from, null)
    }
}
